### PR TITLE
Update HLT and GT packer for new HI L1 objects

### PR DESCRIPTION
**Pull request by apana**
* Update firmware version in cfi for uGT packer (gtStage2Raw_cfi.py)  to pack new HI L1 objects
* Update HLT so TriggerFilterObjectWithRefs objects get filled properly for new HI L1 objects

### CODE REVIEW

**Comment by Martin-Grunewald** [comment]
please test

**Comment by Martin-Grunewald** [comment]
Hi @apana 

What is the type of the new variables? Are they (all?) global event variables (such as ET, ETmiss, HT), ie, one value per event, or could there be several objects of each (or some) variable in an event, (such as there could be several muons in an event, for example)?

I am asking because I need to pack this up at HLT level also...

**Comment by rekovic** [comment]
@Martin-Grunewald  Yes, they are all global event variables (sums), therefore with one object per event.

**Comment by Martin-Grunewald** [comment]
Sorry, but this requires a lot more knock-on code updates on the HLT side, already for the TriggerSummary. But maybe we descope that?

File: EventFilter/L1TRawToDigi/python/gtStage2Raw_cfi.py [hunk 1]
```diff
@@ -12,7 +12,8 @@
     JetInputTag    = cms.InputTag("simCaloStage2Digis"),
     EtSumInputTag  = cms.InputTag("simCaloStage2Digis"),
     FedId = cms.int32(1404),
-    FWId = cms.uint32(0x10A6),  # FW version in GMT with vtx-etrapolation
+    ## FWId = cms.uint32(0x10A6),  # FW version in GMT with vtx-etrapolation
+    FWId = cms.uint32(0x10F2),  # FW version for packing new HI centrality variables
```
Comment (fabiocos) [inline EventFilter/L1TRawToDigi/python/gtStage2Raw_cfi.py]: @apana @rekovic sorry but is this change backward compatible? This fragment is used by top level configurations for many purposes, not just HI. In all the other updates in this PR I see just addition of pieces, which I would assume is backward compatible.

Comment (rekovic) [reply EventFilter/L1TRawToDigi/python/gtStage2Raw_cfi.py]: FwId has increased, and the new version only adds records to pack, not remove any existing records, so should be backward compatible.

**Comment by Martin-Grunewald** [comment]
All these new event variable are stored in the already existing class DataFormats/L1Trigger/interface/EtSum.h, and the added enums flags what the actual variable is?

**Comment by Martin-Grunewald** [comment]
Comments:

1) From the direct comments added by myself to the code changes, it is not clear to me whether some blocks are missing or named differently in different parts of the code changes in this PR, or if anything else is going one in terms of consistency within this PR

2) It was reported by Jaebeom Park that the PR (in 103X) solves the problem he raised originally which led to an earlier and this PR, which is great!

3) As alluded to above, in addition, these new variables should appear in the (HLT/TSG) trigger summary product. This is an additional feature, so this PR can already go in now without waiting for the fix to the trigger summary.

To figure out what is needed for the trigger summary, I need more clarifications, especially on the question:
Are all (10? 12? more?) these new event variables stored in/using the already existing class DataFormats/L1Trigger/interface/EtSum.h, and the added enums are there to flag what the actual meaning of the variable stored is?

**Comment by rekovic** [comment]
Hi @Martin-Grunewald.

1. I believe that @apana covered all the types of L1Triggers that were missing in DataFormats/HLTReco/interface/TriggerTypeDefs.h , some missing from before and some missing now for 2018 HI,  by adding the below.  These are then used to label the products (seeds) stored in the HLTL1TSeed filter.  I don't think that there is any inconsistency there.  I imagine that the problem was not noticed before as none of these types were used in HLT seeding before.
```
     TriggerL1Centrality   =-102,
    TriggerL1MinBiasHFP0  =-103,
    TriggerL1MinBiasHFM0  =-104,
    TriggerL1MinBiasHFP1  =-105,
    TriggerL1MinBiasHFM1  =-106,
    TriggerL1TotalEtEm    =-107,
    TriggerL1MissingHtHF  =-108,
    TriggerL1TowerCount   =-109,
    TriggerL1AsymEt       =-110,
    TriggerL1AsymHt       =-111,
    TriggerL1AsymEtHF     =-112,
    TriggerL1AsymHtHF     =-113,
```
In addition, some of these types, but lesser number,  were also missing from DataFormats/L1TGlobal/src/GlobalObject.cc (for example `TowerCount` used in 2017 HI run, was already implemented).

2. Yes.  The HI checked that the HLTL1TSeed with DiMuon+Centrality triggers is now fine.

3. I agree that the work on TriggerSummary can be postponed.  Yes, the added enums are flagging the new types of L1 EtSums.

**Comment by rekovic** [comment]
+1

**Comment by fabiocos** [comment]
For the record, discussing with @Martin-Grunewald the TriggeSummary will have to be update after 10_3_1 (if this has to be done on 1 day time scale). It might imply an update of the event content, to be clarified

**Comment by fabiocos** [comment]
@Martin-Grunewald if you have no doubts on this PR in itself, are you ready to sign it so as we can proceed with the integration in view of 10_3_1/
And of course also the 10_3_X back-port needs to be approved (also for L1 @rekovic )

**Comment by Martin-Grunewald** [comment]
I would have wished @apana to comment also.

**Comment by Martin-Grunewald** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Comment by apana** [comment]
Hi Martin,

I'm sorry I didn't reply back.  I had thought Vladimir answer was sufficient, but in the future I will reply back to make sure.  Do you still have any questions?  Also, please let me know if you want me to look into updating the TriggerSummary code.

**Comment by Martin-Grunewald** [comment]
Hi Len,

If you look at my specific commenst on the code changes, you see that I am worried that not all the new variables have been added in all locations, as the number of "blocks" added varies.
I interpret Vladimir's feedback as just an assertion/assumption, but you as code author should actually verify - why do you add differently many things in the different locations? And why under different names?

**Pull request merged**